### ISSUE TITLE
Adopt API from Astropy's WCS APE

### ISSUE THREAD

**Issue by Cadair** [sunpy/ndcube#30]
There is an open proposal for a WCS API here: https://github.com/astropy/astropy-APEs/pull/34 I think it would make sense if the `NDCube` API mirror that specification where appropriate, I am specifically suggesting the following methods be changed:

1. Change the signature of `NDCube.pixel_to_world` to:
```
    def pixel_to_world(self, *pixel_arrays):
```
Which would mean accepting arrays of `Quantity` objects in `u.pixel` and dropping the `origin=0` argument (and always assuming `origin=0`).

2. and the same for `NDCube.world_to_pixel`:

```
    def world_to_pixel(self, *world_objects):
```
Which would mean only accepting `SkyCoord` or `Quantity` objects as appropriate (this might be harder in the short-term, we might just have to accept `Quantity`).

3. Change `NDCube.dimensions` so that it just returns the shape of the axes (with units e.g. `GenericMap.dimensions`). Then implement a `NDCube.world_axis_physical_types` method following the APE: (see [this](https://github.com/astrofrog/astropy-APEs/blob/79f6d1eac3cebfce03b8689eb51d39a93d8fd0f7/APE13.rst#common-ucd1-names-for-physical-types))
```
        @property
        def world_axis_physical_types(self):
            """
            Returns an iterable of strings describing the physical type for each
            world axis. They should be names from the VO UCD1+ controlled
            Vocabulary (http://www.ivoa.net/documents/latest/UCDlist.html).
            If no matching UCD type exists, this can instead be "custom:xxx",
            where xxx is an arbitrary string.  Alternatively, if the physical
            type is unknown/undefined, an element can be `None`.
            """
```

**Comment by Cadair** [comment]
I think at least some of this should be done before 0.1. @DanRyanIrish ?

**Comment by DanRyanIrish** [comment]
Hi @Cadair.  Thanks for this issue.  Regarding points 1 and 2, we currently accept a list of ```u.Quantity``` objects for ```pixel_to_world``` and ```world_to_pixel```.  So as far as I see the only change would be getting rid of the ```origin``` kwarg.  You're right thought that we don't currently deal with ```SkyCoord```.  I've never used it before so not sure what is required to support it.

As for point 3, that shouldn't be too hard to implement.

**Comment by DanRyanIrish** [comment]
@Cadair, can you remind me how certain this APE is to be implemented? If it’s guaranteed to be, perhaps we should do it for 1.0. I don’t think it will be doable.

**Comment by DanRyanIrish** [comment]
*do think

**Comment by DanRyanIrish** [comment]
Decision.  For 1.0 we will adopt this API but leave support for ```SkyCoord``` and ```Time``` for a later version.

**Comment by DanRyanIrish** [comment]
Is there an astropy module mapping WCS axis types with the IOVA UCD1+ vocabulary?  Or is there any plan for there to be one?

**Comment by DanRyanIrish** [comment]
This issue is addressed by PR #55
